app/src/main/kotlin/com/wire/android/mapper/MessageMapper.kt

```diff
@@ -114,12 +114,28 @@ class MessageMapper @Inject constructor(
                 messageContent = content,
                 messageSource = if (sender is SelfUser) MessageSource.Self else MessageSource.OtherUser,
                 messageHeader = provideMessageHeader(sender, message),
+                expirationStatus = provideExpirationData(message),
                 messageFooter = footer,
                 userAvatarData = getUserAvatarData(sender)
             )
         }
     }
 
+    private fun provideExpirationData(message: Message.Standalone): ExpirationStatus {
+        val expirationStatus = if (message is Message.Regular) {
+            message.expirationData?.let {
+                ExpirationStatus.Expirable(
+                    it.expireAfter,
+                    it.selfDeletionStatus
+                )
+            } ?: ExpirationStatus.NotExpirable
+        } else {
+            ExpirationStatus.NotExpirable
+        }
+
+        return expirationStatus
+    }
```

Comment:
Nitpick. A bit more readable, perhaps?

```suggestion
    private fun provideExpirationData(message: Message.Standalone): ExpirationStatus {
        val expirationData = (message as? Message.Regular)?.expirationData
        return if (expirationData == null) {
            ExpirationStatus.NotExpirable
        } else {
            ExpirationStatus.Expirable(
                expireAfter = expirationData.expireAfter,
                selfDeletionStatus = expirationData.selfDeletionStatus
            )
        }
    }
```

Reply:
I did not know as? exists, pretty nice thingy, thanks for the tip